src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxQueryBuilder.cs

```diff
@@ -87,14 +85,77 @@ internal class Parameters
         }
     }
 
-    private static IQueryMatch MaterializeWhenNeeded(IQueryMatch source)
+    public struct StreamingOptimization
     {
-        if (source is CoraxBooleanQueryBase cbq)
+        private readonly OrderMetadata _sortField;
+        public readonly bool OptimizationIsPossible;
+        public bool WhereClauseItemMatched;
+
+        public StreamingOptimization(OrderMetadata[] orderMetadata)
         {
-            source = cbq.Materialize();
+            //Conditions to consider streaming optimization:
+            //-sort only by one field
+            //-sort by Sequence/Number.
+            if (orderMetadata is null || orderMetadata.Length > 1 || orderMetadata is not
```

Comment:
Shouldn't this just be: 
```suggestion
            if (orderMetadata is not
```

Reply:
Merged.